### PR TITLE
Eliminate some minor leaks

### PR DESCRIPTION
**Pull request by techee**
Found using the Instruments XCode tool on OS X.

### CODE REVIEW

File: src/highlighting.c [hunk 1]
```diff
@@ -577,7 +577,12 @@ static void styleset_common_init(GKeyFile *config, GKeyFile *config_home)
 	get_keyfile_int(config, config_home, "styling", "line_height",
 		0, 0, &common_style_set.styling[GCS_LINE_HEIGHT]);
 
+	if (&common_style_set.wordchars)
+		g_free(common_style_set.wordchars);
 	get_keyfile_wordchars(config, config_home, &common_style_set.wordchars);
+
```
Comment (b4n) [inline src/highlighting.c]: trailing whitespace

File: src/highlighting.c [hunk 2]
```diff
@@ -577,7 +577,12 @@ static void styleset_common_init(GKeyFile *config, GKeyFile *config_home)
 	get_keyfile_int(config, config_home, "styling", "line_height",
 		0, 0, &common_style_set.styling[GCS_LINE_HEIGHT]);
 
+	if (&common_style_set.wordchars)
```
Comment (b4n) [inline src/highlighting.c]: this test looks off, you don't want to check the address do you?  and there's not need to protect g_free() against NULL either.

File: src/highlighting.c [hunk 3]
```diff
@@ -577,7 +577,12 @@ static void styleset_common_init(GKeyFile *config, GKeyFile *config_home)
...
+		g_free(common_style_set.wordchars);
 	get_keyfile_wordchars(config, config_home, &common_style_set.wordchars);
```
Comment (b4n) [inline src/highlighting.c]: looks like the same fix would be required in `highlighting_init_styles()` (around line 1059), wouldn't it?

File: src/highlighting.c [hunk 4]
```diff
@@ -577,7 +577,12 @@ static void styleset_common_init(GKeyFile *config, GKeyFile *config_home)
...
+	
+	if (whitespace_chars)
+		g_free(whitespace_chars);
```
Comment (b4n) [inline src/highlighting.c]: no need to protect a g_free against NULL

**Comment by techee** [comment]
Thanks for the review, fixed.

**Pull request merged**